modules/core/src/transitions/gpu-interpolation-transition.ts

```diff
@@ -38,120 +27,82 @@ export default class GPUInterpolationTransition implements GPUTransition {
-      const paddedBuffer = padBuffer({buffer, ...padBufferOpts});
-
-      if (buffer !== paddedBuffer) {
-        buffer.destroy();
-        buffers[index] = paddedBuffer;
-
-        // TODO(v9): While this probably isn't necessary as a user-facing warning, it is helpful
-        // for debugging buffer allocation during deck.gl v9 development.
-        console.warn(
-          `[GPUInterpolationTransition] Replaced buffer ${buffer.id} (${buffer.byteLength} bytes) → ` +
-            `${paddedBuffer.id} (${paddedBuffer.byteLength} bytes)`
-        );
-      }
-    }
-
-    this.currentStartIndices = attribute.startIndices;
-    this.currentLength = getAttributeBufferLength(attribute, numInstances);
-    this.attributeInTransition.setData({
-      buffer: buffers[1],
-      // Hack: Float64Array is required for double-precision attributes
-      // to generate correct shader attributes
-      value: attribute.value as NumericArray
+    });
+    buffers[1] = matchBuffer({
+      device: this.device,
+      source: buffers[0],
+      target: buffers[1]
     });
 
-    this.transition.start(transitionSettings);
-
-    this.transform.model.setVertexCount(Math.floor(this.currentLength / attribute.size));
-    // TODO(v9): Best way to handle 'constant' attributes?
-    this.transform.model.setAttributes(
-      attribute.getBuffer() ? {aFrom: buffers[0], aTo: attribute.getBuffer()!} : {aFrom: buffers[0]}
-    );
-    this.transform.transformFeedback.setBuffers({vCurrent: buffers[1]});
+    this.setBuffer(buffers[1]);
+
+    const {transform} = this;
+    const model = transform.model;
+    model.setVertexCount(Math.floor(this.currentLength / attribute.size));
+    if (attribute.isConstant) {
+      model.setAttributes({aFrom: buffers[0]});
+      model.setConstantAttributes({aTo: attribute.value as TypedArray});
+    } else {
+      model.setAttributes({
+        aFrom: buffers[0],
+        aTo: attribute.getBuffer()!
+      });
+    }
+    transform.transformFeedback.setBuffers({vCurrent: buffers[1]});
   }
 
-  update(): boolean {
-    const updated = this.transition.update();
-    if (updated) {
-      const {duration, easing} = this.settings as InterpolationTransitionSettings;
-      const {time} = this.transition;
-      let t = time / duration;
-      if (easing) {
-        t = easing(t);
-      }
-      this.transform.model.setUniforms({time: t});
-      this.transform.run();
+  onUpdate() {
+    const {duration, easing} = this.settings!;
+    const {time} = this.transition;
+    let t = time / duration;
+    if (easing) {
+      t = easing(t);
     }
-    return updated;
+    const {model} = this.transform;
+    model.setUniforms({time: t});
+    // @ts-ignore
+    const gl = model.device.gl as WebGL2RenderingContext;
+    // TODO - remove after https://github.com/visgl/luma.gl/pull/2023
+    gl.bindBuffer(gl.ARRAY_BUFFER, null);
+
+    this.transform.run({
+      clearColor: false,
+      clearDepth: false
+    });
```

Comment:
Resolves the flickering:

```suggestion
    this.transform.run({discard: true});
```

Reply:
Pushed.